presto-main/src/main/java/io/prestosql/server/security/InternalPrincipal.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server.security;
+
+import java.security.Principal;
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+public final class InternalPrincipal
```

Comment:
Does it mean that deserialized `Session` on worker would have coordinator principal (`io.prestosql.Session#identity`)?

Reply:
Oh, I see, original principal goes through `SessionRepresentation`.